### PR TITLE
mirrord: init at 3.125.0

### LINKED ISSUE #305383: Package request: mirrord
**Project description**

Develop Locally with Your Kubernetes Environment

Run your local code in the real-time context of your cloud environment, with access to other microservices, databases, queues, and managed services, all without leaving the local setup you know and love.

**Metadata**

* homepage URL: https://mirrord.dev/
* source URL: https://github.com/metalbear-co/mirrord
* license: mit
* platforms: linux

---

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

CC @aviramha

### PR DESCRIPTION
**Pull request by aaronjheng**
Closes #305383
Closes #310241

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/by-name/mi/mirrord-bin/package.nix [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+  };
+
+  meta = with lib; {
+    description = "Run local processes in the context of Kubernetes environment";
+    homepage = "https://mirrord.dev/";
+    license = licenses.mit;
+    platforms = attrNames manifest.assets;
+    maintainers = with maintainers; [ aaronjheng ];
+    mainProgram = "mirrord";
+  };
```
Comment (Aleksanaa) [inline pkgs/by-name/mi/mirrord-bin/package.nix:53]:
```suggestion
  meta = {
    description = "Run local processes in the context of Kubernetes environment";
    homepage = "https://mirrord.dev/";
    license = lib.licenses.mit;
    platforms = attrNames manifest.assets;
    maintainers = with lib.maintainers; [ aaronjheng ];
    mainProgram = "mirrord";
  };
```

File: pkgs/by-name/mi/mirrord-bin/package.nix [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+    updateScript = ./update.py;
+  };
+
+  meta = with lib; {
+    description = "Run local processes in the context of Kubernetes environment";
+    homepage = "https://mirrord.dev/";
+    license = licenses.mit;
+    platforms = attrNames manifest.assets;
+    maintainers = with maintainers; [ aaronjheng ];
+    mainProgram = "mirrord";
```
Comment (Aleksanaa) [inline pkgs/by-name/mi/mirrord-bin/package.nix:52]:
```suggestion
    mainProgram = "mirrord";
    sourceProvenance = with lib.sourceTypes; [ binaryNativeCode ];
```

File: pkgs/by-name/mi/mirrord-bin/package.nix [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+  fetchurl,
+  autoPatchelfHook,
+  testers,
+  mirrord-bin,
+}:
+
+let
+  manifest = lib.importJSON ./manifest.json;
+in
+stdenv.mkDerivation (finalAttrs: {
```
Comment (Aleksanaa) [inline pkgs/by-name/mi/mirrord-bin/package.nix:13]:
```suggestion
stdenvNoCC.mkDerivation (finalAttrs: {
```

Comment (aaronjheng) [reply pkgs/by-name/mi/mirrord-bin/package.nix:13]: "Although I don't know the specific reason, stdenv is required here. See `buildInputs`.

File: pkgs/by-name/mi/mirrord/package.nix [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+in
+stdenv.mkDerivation (finalAttrs: {
+  pname = "mirrord";
+  version = manifest.version;
+
+  src =
+    let
+      system = stdenv.hostPlatform.system;
+    in
+    fetchurl (manifest.assets.${system} or (throw "Unsupported system: ${system}"));
```
Comment (pluiedev) [inline pkgs/by-name/mi/mirrord/package.nix:21]: It won't evaluate for platforms that are not listed in `meta.platforms` anyway.
```suggestion
    fetchurl (manifest.assets.${system});
```

File: pkgs/by-name/mi/mirrord/package.nix [hunk 5]
```diff
@@ -0,0 +1,55 @@
...
+    updateScript = ./update.py;
+  };
+
+  meta = {
+    description = "Run local processes in the context of Kubernetes environment";
+    homepage = "https://mirrord.dev/";
+    license = lib.licenses.mit;
+    platforms = builtins.attrNames manifest.assets;
+    maintainers = with lib.maintainers; [ aaronjheng ];
+    sourceProvenance = with lib.sourceTypes; [ binaryNativeCode ];
```
Comment (pluiedev) [inline pkgs/by-name/mi/mirrord/package.nix:52]: It shouldn't be *that* difficult to build from source, right?
https://github.com/metalbear-co/mirrord
https://github.com/metalbear-co/mirrord/blob/main/CONTRIBUTING.md

Comment (aaronjheng) [reply pkgs/by-name/mi/mirrord/package.nix:52]: I'm not sure. Just want to push forward #310241. I hope to try building from source in the future.

**Comment by aaronjheng** [comment]
Thanks!

**Pull request merged**